Samples/Samples/ViewModel/CalendarViewModel.cs

```diff
@@ -11,28 +13,77 @@ public class CalendarViewModel : BaseViewModel
         public CalendarViewModel()
         {
             GetCalendars = new Command(OnClickGetCalendars);
+            GetAllEvents = new Command(OnClickGetEvents);
             RequestCalendarReadAccess = new Command(OnRequestCalendarReadAccess);
             RequestCalendarWriteAccess = new Command(OnRequestCalendarWriteAccess);
         }
 
         public ICommand GetCalendars { get; }
 
+        public ICommand GetAllEvents { get; }
+
         public ICommand RequestCalendarReadAccess { get; }
 
         public ICommand RequestCalendarWriteAccess { get; }
 
         public ObservableCollection<ICalendar> Calendars { get; } = new ObservableCollection<ICalendar>();
 
+        public ObservableCollection<IEvent> Events { get; } = new ObservableCollection<IEvent>();
+
+        ICalendar selectedCalendar;
+
+        public ICalendar SelectedCalendar
+        {
+            get
+            {
+                return selectedCalendar;
+            }
+
+            set
+            {
+                if (selectedCalendar != value)
+                {
+                    selectedCalendar = value;
+                    OnPropertyChanged("SelectedCalendar");
+                    if (selectedCalendar != null)
+                    {
+                        OnChangeRequestCalendarSpecificEvents(selectedCalendar.Id);
+                    }
+                }
+            }
+        }
+
         async void OnClickGetCalendars()
         {
             Calendars.Clear();
+            Calendars.Add(new DeviceCalendar() { Id = null, IsReadOnly = true, Name = "All" });
             var calendars = await Calendar.GetCalendarsAsync();
             foreach (var calendar in calendars)
             {
                 Calendars.Add(calendar);
             }
         }
 
+        async void OnChangeRequestCalendarSpecificEvents(string calendarId)
+        {
+            Events.Clear();
+            var events = await Calendar.GetEventsAsync(calendarId);
```

Comment:
If calendarId==null will this return events across all calendars, or should it be calling OnClickGetEvents for that?

Reply:
Removing method OnClickGetEvents as it was used for the other/deprecated implementation of xaml.

As the method handles a null parameter as getting all, I will alter the function to:
async void OnChangeRequestCalendarSpecificEvents(string calendarId = null)

To make it more obvious, I can add a comment as well if this is still not clear? But if I were should I add that above this function, or within the Calendar class, since this is only a sample app so if people were to f12 into the function to understand what it was doing I'm guessing they would be taken to the Calendar.shared.cs implementation, and thus it may be better to have a comment in there.